runtime/near-vm-logic/src/logic.rs

```diff
@@ -637,6 +639,33 @@ impl<'a> VMLogic<'a> {
         Ok(self.context.block_index)
     }
 
+    /// Gets a block given a height (up to 256)
+    ///
+    /// # Cost
+    ///
+    /// `base`
+    // TODO #1903 rename to `block_height`
+    pub fn get_block_by_hash(&mut self, height_number: BlockHeight) -> Result<u64> {
+        self.gas_counter.pay_base(base)?;
+        let chain_store = self.context.chain_store;
+        let tip = chain_store.header_head().expect("handle this properly");
+        if tip.height == height_number {
+            return Ok(1);
+        }
+        let mut previous_block_hash = tip.prev_block_hash;
+        // iterate up to 255 times more
+        for _ in 0..255 {
+            let block_header =
+                chain_store.get_block_header(&previous_block_hash).expect("handle this properly");
+            if block_header.height() == height_number {
+                return Ok(1);
+            }
```

Comment:
Heights aren’t continuous so it might be beneficial to handle skipped
heights here.  Something like:

```rust
match block_header.height().cmp(&height_number) {
    std::cmp::Ordering::Less => {
        // User requested block height which does not exist, i.e. a block height
        // has been skipped
        return Ok(2);
    },
    std::cmp::Ordering::Equal => {
        // TODO: return the hash to the user
        return Ok(1);
    },
    std::cmp::Ordering::Greater => {
        // Still need to continue going back.
    }
}
```